scrapy/resolver.py

```diff
@@ -16,8 +16,7 @@ def __init__(self, reactor, cache_size, timeout):
     def getHostByName(self, name, timeout=None):
         if name in dnscache:
             return defer.succeed(dnscache[name])
-        if not timeout:
-            timeout = self.timeout
+        timeout = (self.timeout,)
```

Comment:
it means getHostByName callers can't not override timeout which is acceptable but it is possible for `self.timeout` to be a tuple and convert it only if not. This way we don't lost access to existent functionality provided by `t.i.b.ThreadedResolver.`